nexus/db-queries/src/db/datastore/volume.rs

```diff
@@ -688,15 +686,32 @@ impl DataStore {
 
                     // A volume (for a disk or snapshot) may reference another nested
                     // volume as a read-only parent, and this may be arbitrarily deep.
-                    // After decrementing volume_references above, get all region
-                    // snapshot records where the volume_references has gone to 0.
-                    // Consumers of this struct will be responsible for deleting the
-                    // read-only downstairs running for the snapshot and the snapshot
-                    // itself.
+                    // After decrementing volume_references above, get the region
+                    // snapshot records for these read_only_targets where the
+                    // volume_references has gone to 0. Consumers of this struct will
+                    // be responsible for deleting the read-only downstairs running
+                    // for the snapshot and the snapshot itself.
                     datasets_and_snapshots: {
                         use db::schema::dataset::dsl as dataset_dsl;
 
                         dsl::region_snapshot
+                            // Only return region_snapshot records related to
+                            // this volume that have zero references. This will
+                            // only happen one time, on the last decrease of a
+                            // volume containing these read-only targets.
+                            //
+                            // It's important to not return *every* region
+                            // snapshot with zero references: multiple volume
+                            // delete sub-sagas will then be issues duplicate
+                            // DELETE calls to Crucible agents, and a request to
+                            // delete a read-only downstairs running for a
+                            // snapshot that doesn't exist will return a 404,
+                            // causing the saga to error and unwind.
+                            .filter(
```

Comment:
So down on line 735ish is where we actually store the "stuff to be cleaned up" -- in the case that multiple snapshots are sharing resources for a volume, it's basically the responsibility of the "last one deleted" to set this field with the set of regions to be deleted, right?